Python code incorrectly assumes that all negative values returned by the
sched_get_priority calls are error indicators; however that is not true
in Solaris (where some policies return -60 as a valid value).

This was reported and patch offered upstream:
https://bugs.python.org/issue41839

--- Python-3.9.0/Modules/posixmodule.c
+++ Python-3.9.0/Modules/posixmodule.c
@@ -6283,7 +6283,7 @@ os_sched_get_priority_max_impl(PyObject
     int max;
 
     max = sched_get_priority_max(policy);
-    if (max < 0)
+    if (max == -1)
         return posix_error();
     return PyLong_FromLong(max);
 }
@@ -6302,7 +6302,7 @@ os_sched_get_priority_min_impl(PyObject
 /*[clinic end generated code: output=7595c1138cc47a6d input=21bc8fa0d70983bf]*/
 {
     int min = sched_get_priority_min(policy);
-    if (min < 0)
+    if (min == -1)
         return posix_error();
     return PyLong_FromLong(min);
 }
